Add test for PredicateContainer

jevan13 opened the pull request:
## Describe your changes
- Improve code coverage
- Add test for equals and getCombinedPredicates() to check for null Predicates

## Issue ticket number and link
closes #128 

## Checklist before requesting a review
- [X] I have performed a self-review of my code.
- [X] If it is a core feature, I have added thorough tests.

jevan13 committed:
Add test for PredicateContainer

- Improve code coverage
- Add test for equals and getCombinedPredicates() to check for null Predicates

LimErnest commented on src/test/java/seedu/address/model/person/PredicateContainerTest.java:
```diff
@@ -16,51 +17,12 @@
-                .addEmailContainsKeywordsPredicate(emailPredicate1)
-                .addDepartmentContainsKeywordsPredicate(departmentPredicate1)
-                .addRoleContainsKeywordsPredicate(rolePredicate1);
+        PredicateContainer predicateContainer3 = getPredicateContainer1();
```
This test might be the same as the one above where you are testing for the same object

jevan13 replied:
The other tests for the same instance (predicateContainer1 against predicateContainer1), whereas this tests for different instances (predicateContainer1 against predicateContainer3), where both predicateContainer contains the same predicates.

LimErnest replied:
I see now, thanks for the explanation!

LimErnest commented on src/test/java/seedu/address/model/person/PredicateContainerTest.java:
```diff
@@ -147,5 +82,51 @@ public void test_toString() {
         assertEquals(expected, actual);
     }
 
+    public PredicateContainer getPredicateContainer1() {
```
Could add comments explaining what getPredicateContainter 1 and 2 do

jevan13 replied:
Added!

jevan13 committed:
Add JavaDocs for getPredicateContainer1() and getPredicateContainer2()

- Add JavaDocs to explain the difference between the two PredicateContainers
- Improve understanding of test cases

LimErnest reviewed (approved):
LGTM!
